src/modules/billing/services/batch-service.js

```diff
@@ -350,26 +352,19 @@ const deleteBatchInvoice = async (batch, invoiceId) => {
   if (!invoice) {
     throw new NotFoundError(`Invoice ${invoiceId} not found`);
   }
-
   // Set batch status back to 'processing'
   await setStatus(batch.id, Batch.BATCH_STATUS.processing);
 
   try {
-    await chargeModuleBillRunConnector.deleteInvoiceFromBillRun(invoice.billingBatch.externalId, invoice.externalId);
-    // Delete local data
-    await newRepos.billingBatchChargeVersionYears.deleteByInvoiceId(invoiceId);
-    await newRepos.billingVolumes.deleteByBatchAndInvoiceId(batch.id, invoiceId);
-    await newRepos.billingTransactions.deleteByInvoiceId(invoiceId);
-    await newRepos.billingInvoiceLicences.deleteByInvoiceId(invoiceId);
-
-    // Delete the invoice if the rebilling state is null otherwise
-    // reset the original invoice and delete the rebilled invoices.
-    if (invoice.rebillingState === null) {
-      await newRepos.billingInvoices.delete(invoiceId);
-    } else {
-      await deleteRebillingInvoices(invoice.originalBillingInvoiceId);
+    const invoicesToDelete = [invoice];
+
+    if (invoice.originalBillingInvoiceId !== null) {
+      await invoiceService.updateInvoice(invoice.originalBillingInvoiceId, { isFlaggedForRebilling: false, originalBillingInvoiceId: null, rebillingState: null });
+      invoicesToDelete.push([...invoice.linkedBillingInvoices].filter(row => row.billingInvoiceId !== row.originalBillingInvoiceId));
```

Comment:
I think this may end up attempting to delete one of the pair twice.  This is because the base invoice is only filtered out from `linkedBillingInvoices` at the mapping layer, and this code is using the repo layer directly.

So I think we possibly need to just overwrite invoicesToDelete:
```
 const isNotOriginalInvoice = row => row.billingInvoiceId !== row.originalBillingInvoiceId);

// ...

     invoicesToDelete = invoice.linkedBillingInvoices.filter(isNotOriginalInvoice);
```